### PR TITLE
Feat/541/vidvox isf modules

### LINKED ISSUE #541: Add all ISF modules from Vidvox into modV
All the modules from https://github.com/Vidvox/ISF-Files/tree/master/ISF are under MIT and we could add them into modV.

### PR DESCRIPTION
**Pull request by TimPietrusky**
fixes #541

Not all modules are added, see https://github.com/vcync/modV/issues/541#issuecomment-869228008

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -6024,9 +6024,13 @@ inquirer@^6.2.2:
     strip-ansi "^5.1.0"
     through "^2.3.6"
 
-"interactive-shader-format@github:vcync/interactive-shader-format-js#feat/ImageBitmap":
+"interactive-shader-format@link:../../../../../Users/timpietrusky/dev/js/vcync/interactive-shader-format-js":
```
Comment (2xAA) [inline yarn.lock]: This doesn't look right 🤔

File: src/application/worker/loop.js [hunk 2]
```diff
@@ -149,7 +149,11 @@ function loop(delta, features) {
 
       const module = active[group.modules[j]];
 
-      if (!module.meta.enabled || module.meta.alpha < 0.001) {
+      if (
+        module === undefined ||
```
Comment (2xAA) [inline src/application/worker/loop.js]: This is probably an okay check, but we really shouldn't be encountering an undefined module here.
What's the case that prompted the check to be added here? We should fix the cause.

File: src/application/renderers/isf.js [hunk 3]
```diff
@@ -87,6 +85,10 @@ async function setupModule(module) {
   const renderer = new ISFRenderer(isfContext);
   renderer.loadSource(fragmentShader, vertexShader);
 
+  if (!renderer.valid) {
+    throw renderer.errorWithCorrectedLines;
```
Comment (2xAA) [inline src/application/renderers/isf.js:89]: Is `errorWithCorrectedLines` new? I've not seen this before.

File: src/application/worker/store/modules/modules.js [hunk 4]
```diff
@@ -325,6 +325,11 @@ const actions = {
     { state, commit },
     { moduleId, prop, data, group, groupName, writeToSwap }
   ) {
+    if (state.active[moduleId] === undefined) {
```
Comment (2xAA) [inline src/application/worker/store/modules/modules.js]: I'd probably use `if (!state.active[moduleId]) {` here as that's more consistent with the rest of the codebase - most of the time we don't really care if it's actually undefined, we just need to know something is there.

**Review by 2xAA — CHANGES_REQUESTED** [review]
It's been a long time coming - so happy this PR is finally here 😄 
Looks like some great additions here. Just a few comments from me, nothing major.

Great work, @TimPietrusky!

File: src/application/worker/loop.js (hunk 2 above)
Comment (TimPietrusky) [reply src/application/worker/loop.js]: I tested this again and the problem comes up when two modules share the same name.

## Steps to reproduce
* Create a module called `Lines` in the in `modV/media/default/module` of type `shader`
* Load the branch `feat/541/vidvoxIsfModules` which contains an ISF module called `Lines`
* See these errors:

```
modules.js#removeActiveModule: Cannot destructure property 'meta' of 'module' as it is undefined.

loop.js: Cannot read property 'meta' of undefined

loop.js: Cannot destructure property 'meta' of 'module' as it is undefined.

modules.js#init: Cannot read property '$moduleName' of undefined
```

Not all of them are shown all the time, but if the module-name-conflict is resolved, none of them are shown. 

What I also see is this warning a couple of times:

```
Did not remove inputLink. Could not find input with id
```

So I think this is a new bug and I will remove the check for `undefined` from this PR, what do you think @2xAA ?

File: src/application/renderers/isf.js (hunk 3 above)
Comment (TimPietrusky) [reply src/application/renderers/isf.js:89]: Yes, I added this to ISFRenderer as the original was just giving back the errorLine of the first error and the error-message still contains the errorLines from the parsed shader, which is not helpful at all. 

See https://github.com/vcync/interactive-shader-format-js/blob/fixedIsfParserAndImageBitmap/src/ISFRenderer.js#L55

File: src/application/worker/store/modules/modules.js (hunk 4 above)
Comment (TimPietrusky) [reply src/application/worker/store/modules/modules.js]: Fixed!

File: yarn.lock (hunk 1 above)
Comment (TimPietrusky) [reply yarn.lock]: Fixed!

File: src/application/worker/loop.js (hunk 2 above)
Comment (2xAA) [reply src/application/worker/loop.js]: @TimPietrusky Yes I agree.

This sounds like a bug in the module creation. I'm surprised naming conflicts still happen, but let's create a new ticket as you suggest and remove the check for undefined here.

**Review by 2xAA — APPROVED** [review]

**Review by 2xAA — CHANGES_REQUESTED** [review]
Remove `module === undefined` in `loop.js` as discussed and we can merge after the new naming conflict bug has been resolved.

**Comment by TimPietrusky** [comment]
@2xAA the change was removed and the bug opened: https://github.com/vcync/modV/issues/600

**Review by 2xAA — APPROVED** [review]

File: src/application/worker/loop.js (hunk 2 above)
Comment (TimPietrusky) [reply src/application/worker/loop.js]: See #600

File: src/application/worker/store/modules/modules.js (hunk 4 above)
Comment (TimPietrusky) [reply src/application/worker/store/modules/modules.js]: This was also done!

**Pull request merged**